torch/onnx/_internal/fx/frontend.py

```diff
@@ -0,0 +1,286 @@
+from __future__ import annotations
+
+import abc
+import copy
+import functools
+
+import inspect
+
+from typing import Any, Callable, Dict, Optional, Tuple
+
+import torch
+import torch._dynamo
+import torch.fx
+from torch._functorch import aot_autograd
+from torch.fx.experimental import proxy_tensor
+from torch.onnx._internal import _beartype
+
+DecompositionTableType = Dict[Callable, Callable]
+
+
+class FxFrontend(abc.ABC):
+    """Base class for all FX frontends.
+
+    Providing a common interface for all FX frontends. To allow onnx exporter to easily
+    experiment with and swap between different frontends.
+    """
+
+    @_beartype.beartype
+    def __init__(self, *config_args, **config_kwargs):
+        """Initialize the frontend with config arguments."""
+        ...
+
+    @abc.abstractmethod
+    @_beartype.beartype
+    def _inner_call(self, model: Callable, *args, **kwargs) -> torch.fx.GraphModule:
+        ...
+
+    @_beartype.beartype
+    def __call__(self, model: Callable, *args, **kwargs) -> torch.fx.GraphModule:
```

Comment:
Perhaps `def trace` is better?

Reply:
Yes, let's be explicit in code. My first impression to `FxFrontend.__calll__` or `FxFrontend.__calll__` is a question mark. If it's named after what that function really does, it'd be easier to get a quick understanding.